src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java

```diff
@@ -15,54 +15,56 @@
 package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
 
 public class InputUnnecessaryParenthesesCheckSwitchExpression {
-    MathOperation2 tooManyParens(int k) {
-        return switch (k) {
-            case 1 -> {
-                MathOperation2 case5 = (a, b) -> (a + b); // violation
-                yield case5;
-            }
-            case (2) -> { // violation
-                MathOperation2 case6 = (int a, int b) -> (a + b); // violation
-                yield case6;
-            }
-            case 3 -> {
-                MathOperation2 case7 = (int a, int b) -> {
-                    return (a + b); // violation
-                };
-                yield (case7); // violation
-            }
-            default -> {
-                MathOperation2 case8 = (int x, int y) -> {
-                    return (x + y); // violation
-                };
-                yield case8;
-            }
+  MathOp2 tooManyParens(int k) {
+    return switch (k) {
+      case 1 -> {
+        MathOp2 e = (a, b) -> (a + b); // violation '.* parentheses around assignment right.*side'
+        yield e;
```

Comment:
No need for leading or trailing `.*` it gets added automatically.
`.* parentheses around assignment right.*side` 
↓
 `parentheses around assignment right.*side`
Please change it for all the lines.

Reply:
@faraazb , ∆∆

Reply:
Done

Reply:
Resolved.